mocca-client/src/main/java/com/paypal/mocca/client/MoccaExceptionHandler.java

```diff
@@ -0,0 +1,38 @@
+package com.paypal.mocca.client;
+
+import feign.FeignException;
+
+import javax.validation.ConstraintViolationException;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * This exception handler allows Mocca to hide Feign exception types from applications
+ *
+ * @author facarvalho
+ */
+final class MoccaExceptionHandler {
+
+    private MoccaExceptionHandler() {
+    }
+
+    private static final Set<Class<? extends Throwable>> acceptableExceptions = new HashSet<>();
+
+    static {
+        acceptableExceptions.add(MoccaException.class);
+        acceptableExceptions.add(ConstraintViolationException.class);
+    }
+
+    static Throwable handleException(Throwable throwable) {
```

Comment:
IMO, we should unit test for this.  I don't know that our functional testing will be sufficient to uncover some cases we'd like to cover (i.e. not wrapping Error).  But, I know there are different stances on what 'unit testing' looks like.  I leave it up to you:)

Reply:
Added unit test